app/Lib/Tools/BackgroundJobsTool.php

```diff
@@ -0,0 +1,681 @@
+     */
+    private function validateQueue(string $queue): bool
+    {
+        if (!in_array($queue, self::VALID_QUEUES)) {
+            throw new InvalidArgumentException(
+                sprintf(
+                    'Invalid background job queue, must be one of: [%s]',
+                    implode(', ', self::VALID_QUEUES)
+                )
+            );
+        }
+
+        return true;
+    }
+
+    /**
+     * Validate command
+     *
+     * @return boolean
+     * @throws InvalidArgumentException
+     */
+    private function validateCommand(string $command): bool
+    {
+        if (!in_array($command, self::ALLOWED_COMMANDS)) {
+            throw new InvalidArgumentException(
+                sprintf(
+                    'Invalid command, must be one of: [%s]',
+                    implode(', ', self::ALLOWED_COMMANDS)
+                )
+            );
+        }
+
+        return true;
+    }
+
+    /**
+     * Validate worker name
+     *
+     * @return boolean
+     * @throws InvalidArgumentException
+     */
+    private function validateWorkerName(string $name): bool
+    {
+        list($queue, $id) = explode('_', $name);
+
+        $this->validateQueue($queue);
+
+        if (!$this->validateQueue($queue) || !is_numeric($id)) {
+            throw new InvalidArgumentException('Invalid worker name, must be one of format {queue_name}_{process_id}, example: default_00');
+        }
+
+        return true;
+    }
+
+    /**
+     * @return Redis
+     */
+    private function createRedisConnection(): Redis
+    {
+        $redis = new Redis();
+        $redis->connect($this->settings['redis_host'], $this->settings['redis_port']);
+        $redisPassword = $this->settings['redis_password'];
+
+        if (!empty($redisPassword)) {
+            $redis->auth($redisPassword);
+        }
+        $redis->select($this->settings['redis_database']);
+
+        return $redis;
+    }
+
+    /**
+     * @return \Supervisor\Supervisor
+     */
+    private function createSupervisorConnection(): \Supervisor\Supervisor
+    {
+        $httpOptions = [];
+        if (!empty($this->settings['supervisor_user']) && !empty($this->settings['supervisor_password'])) {
+            $httpOptions = [
+                'auth' => [
+                    $this->settings['supervisor_user'],
+                    $this->settings['supervisor_password'],
+                ],
+            ];
+        }
+
+        $client = new \fXmlRpc\Client(
+            sprintf(
+                'http://%s:%s/RPC2',
+                $this->settings['supervisor_host'],
+                $this->settings['supervisor_port']
+            ),
+            new \fXmlRpc\Transport\HttpAdapterTransport(
+                new \Http\Message\MessageFactory\GuzzleMessageFactory(),
+                new \GuzzleHttp\Client($httpOptions)
+            )
+        );
+
+        return new \Supervisor\Supervisor($client);
+    }
+
+    private function getJobById(int $jobId): ?Job
```

Comment:
What about to change this method to directly save process ID? Because it is only use here.